### PR TITLE
Vhbb/add jetsd r08 to higgs candidate

### PR DESCRIPTION
**Pull request by degrutto**
should replace #101 

also added a first trial of vhbb_data

### CODE REVIEW

File: VHbbAnalysis/Heppy/python/VHbbAnalyzer.py [hunk 1]
```diff
@@ -178,26 +178,28 @@ def doHiggsHighPt(self,event) :
...
-           event.hJets3cj.append(j3)
-#          event.hJets3cj=list(min(itertools.combinations(event.jetsForHiggs,3), key = lambda x : ( x[2]>15 and (deltaR( x[0].eta(), x[0].phi(), x[2].eta(), x[2].phi()) +  deltaR( x[1].eta(), x[1].phi(), x[2].eta(), x[2].phi()) ) ) ))
-           event.aJets3cj = [x for x in event.cleanJets if x not in event.hJets3cj]
-           event.hjidx3cj=[event.cleanJetsAll.index(x) for x in event.hJets3cj ]
-           event.ajidx3cj=[event.cleanJetsAll.index(x) for x in event.aJets3cj ]
-           event.aJets3cj+=event.cleanJetsFwd
-           event.H3cj = event.hJets3cj[0].p4()+event.hJets3cj[1].p4()+event.hJets3cj[2].p4()
-           event.minDr3 = min(deltaR( event.hJets3cj[0].eta(), event.hJets3cj[0].phi(), event.hJets3cj[2].eta(), event.hJets3cj[2].phi()) ,  deltaR( event.hJets3cj[1].eta(), event.hJets3cj[1].phi(), event.hJets3cj[2].eta(), event.hJets3cj[2].phi()))  
+    def doHiggsAddJetsdR08(self,event) :
+        event.hJetsaddJetsdR08 = [x for x in event.hJetsCSV]
```
Comment (arizzi) [inline VHbbAnalysis/Heppy/python/VHbbAnalyzer.py]: can you explain what is this meant for? why do you need a copy of all these collections?!

**Comment by degrutto** [comment]
Hi Andrea,

I did these in order to speed the process and loop only on a subset of all jets, 
please modify of you believe it is still not optimal

   Michele

**Comment by arizzi** [comment]
but why you copy them? can't you just use event.hJetsCSV ?

**Comment by degrutto** [comment]
the thing is I am going to append them

event.hJetsaddJetsdR08.append(x) 
later

I would keep it as a separate collection

         Michele

**Comment by arizzi** [comment]
how about the others? aJets?
Wouldn't it be better to just have the indices of the matched jets?

**Comment by degrutto** [comment]
Hi Andrea,
sorry to reply onl now,

so far the logic I was following is 

to have a new collections of   event.hJetsaddJetsdR08 which contains always the two higgs jets (high csv) and any of the matched jets and then save the index of these guys. The reconstructed higgs would then be the two HCSV plus eventually the additional jets. So in the analysis one could easier start for all events from reconstructed higgs HaddJetsdR08 and then lopping on the indexes hjidxaddJetsdR08 could navigate the two highest csv plus eventually the added FSR jets.
To speed the things I was creating intermediate (and not saved) collections such as addJetsForHiggs to limit the loop two jets with pt>15 and eta<3.0 and pu id.

Indeed, as you suggest we can instead just save the indexes of the added FSR jets and leave the analysts to do later the jets recombination, but that would descope the things I fear.

If you have strong opinion code wise or performance wise ok,
but if not I believe this is the simplest way to save these extra information and make all use them coherently.

cheers

    Michele

**Pull request merged**